src/js/_enqueues/admin/site-icon.js

```diff
@@ -0,0 +1,152 @@
+(function($) {
+	var frame;
+
+	function calculateImageSelectOptions ( attachment ) {
+		var realWidth  = attachment.get( 'width' ),
+			realHeight = attachment.get( 'height' ),
+			xInit = 512,
+			yInit = 512,
+			ratio = xInit / yInit,
+			xImg  = xInit,
+			yImg  = yInit,
+			x1, y1, imgSelectOptions;
+
+		if ( realWidth / realHeight > ratio ) {
+			yInit = realHeight;
+			xInit = yInit * ratio;
+		} else {
+			xInit = realWidth;
+			yInit = xInit / ratio;
+		}
+
+		x1 = ( realWidth - xInit ) / 2;
+		y1 = ( realHeight - yInit ) / 2;
+
+		imgSelectOptions = {
+			aspectRatio: xInit + ':' + yInit,
+			handles: true,
+			keys: true,
+			instance: true,
+			persistent: true,
+			imageWidth: realWidth,
+			imageHeight: realHeight,
+			minWidth: xImg > xInit ? xInit : xImg,
+			minHeight: yImg > yInit ? yInit : yImg,
+			x1: x1,
+			y1: y1,
+			x2: xInit + x1,
+			y2: yInit + y1
+		};
+
+		return imgSelectOptions;
+	}
+
+	$( function() {
+		// Build the choose from library frame.
+		$( '#choose-from-library-link' ).on( 'click', function( event ) {
+			var $el = $(this);
+			event.preventDefault();
```

Comment:
By making sure the button has a `type="button"` attribute, there's no default action to prevent and this line can be removed.